image/descriptor.go

```diff
@@ -116,15 +115,18 @@ func (d *descriptor) validate(w walker, mts []string) error {
 }
 
 func (d *descriptor) validateContent(r io.Reader) error {
-	h := sha256.New()
-	n, err := io.Copy(h, r)
+	parsed, err := digest.Parse(d.Digest)
```

Comment:
I'd love to (and I do that in #5).  However, [as I pointed out earlier][1], that would require making this PR larger (and therefore harder to review).  I'll gladly file a follow up PR with that change if we can land this initial go-digest consuming code first.

[1]: https://github.com/opencontainers/image-tools/pull/104#issuecomment-274305024